config/private-registry.yml

```diff
@@ -0,0 +1,71 @@
+#@ load("@ytt:overlay", "overlay")
+#@ load("@ytt:base64", "base64")
+#@ load("@ytt:json", "json")
+#@ load("@ytt:data", "data")
+#@ load("/namespaces.star", "system_namespace", "workloads_namespace", "workloads_staging_namespace")
+
+#! add dockerconfigjson secret to all namespaces to use as imagePullSecret in ServiceAccounts and Pod specs
+
+#@ if data.values.private_registry.add_image_pull_secrets:
+#@ namespaces = [ system_namespace(), workloads_namespace(), workloads_staging_namespace(), "cf-db", "cf-blobstore", "kpack", "istio-system" ]
+#@ for namespace in namespaces:
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: private-registry-auth-secret
+  namespace: #@ namespace
+type: kubernetes.io/dockerconfigjson
+data:
+  #@ docker_auth = base64.encode("{}:{}".format(data.values.private_registry.username, data.values.private_registry.password))
+  #@ docker_creds = {"username": data.values.private_registry.username, "password": data.values.private_registry.password, "auth": docker_auth}
+  .dockerconfigjson: #@ base64.encode(json.encode({"auths": {data.values.private_registry.hostname: docker_creds}}))
+#@ end
+
+#@overlay/match by=overlay.subset({"kind": "ServiceAccount"}), expects="1+"
+---
+#@overlay/match missing_ok=True
+imagePullSecrets:
+#@overlay/append
+- name: private-registry-auth-secret
+
+#@overlay/match by=overlay.subset({"kind": "Deployment"}), expects="0+"
+---
+spec:
+  template:
+    spec:
+      #@overlay/match missing_ok=True
+      imagePullSecrets:
+      #@overlay/append
+      - name: private-registry-auth-secret
+
+#@overlay/match by=overlay.subset({"kind": "StatefulSet"}), expects="0+"
+---
+spec:
+  template:
+    spec:
+      #@overlay/match missing_ok=True
+      imagePullSecrets:
+      #@overlay/append
+      - name: private-registry-auth-secret
+
+#@overlay/match by=overlay.subset({"kind": "DaemonSet"}), expects="0+"
+---
+spec:
+  template:
+    spec:
+      #@overlay/match missing_ok=True
+      imagePullSecrets:
+      #@overlay/append
+      - name: private-registry-auth-secret
+
+#@overlay/match by=overlay.subset({"kind": "Job"}), expects="0+"
+---
+spec:
+  template:
+    spec:
+      #@overlay/match missing_ok=True
+      imagePullSecrets:
+      #@overlay/append
+      - name: private-registry-auth-secret
+#@ end
```

Comment:
Could these be collapsed into a single overlay using the `overlay.or_op()` [helper](https://github.com/k14s/ytt/blob/develop/docs/lang-ref-ytt-overlay.md#overlayor_op)?

Reply:
Oh nice, I didn't know about `overlay.or_op()`. 
Yes of course, changed it. 👍